### PR TITLE
Transaction mediator integration tests

### PR DESCRIPTION
**Pull request by prabushi**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    private static final String COMMIT_CONTEXT = "commit";
+    private static final String ROLLBACK_CONTEXT = "rollback";
+    private static final String TEST_CONTEXT = "test";
+    private static final String CLEANUP_CONTEXT = "cleanup";
+    private static final String FAULT_NOTX_CONTEXT = "fault-if-no-tx";
+    private static final String USE_EXISTING_NEW_CONTEXT = "use-existing-or-new";
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        HttpRequestUtil.doPost(new URL(API_URL + INIT_CONTEXT), "");
```
Comment (milindaperera) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java]: Why are we performing test work here? 
Do only environment initialization work here

Comment (prabushi) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java]: We need to initialize database(create tables and relevant db entries) before the following tests. Is there any other common place to perform those actions?

Comment (milindaperera) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java]: Ah .. if it is the intention, then this is fine. Shall we add a comment here expressing the intention of this API call

Comment (prabushi) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/transaction/TransactionMediatorTestCase.java]: Sure, will add a comment.

**Pull request merged**